app/forms/candidate_interface/english_gcse_grade_form.rb

```diff
@@ -0,0 +1,278 @@
+module CandidateInterface
+  class EnglishGcseGradeForm
+    include ActiveModel::Model
+    include ValidationUtils
+
+    attr_accessor :grade,
+                  :award_year,
+                  :qualification,
+                  :other_grade,
+                  :grades,
+                  :english_gcses,
+                  :english_single_award,
+                  :grade_english_single,
+                  :english_double_award,
+                  :grade_english_double,
+                  :english_language,
+                  :grade_english_language,
+                  :english_literature,
+                  :grade_english_literature,
+                  :english_studies_single_award,
+                  :grade_english_studies_single,
+                  :english_studies_double_award,
+                  :grade_english_studies_double,
+                  :other_english_gcse,
+                  :other_english_gcse_name,
+                  :grade_other_english_gcse
+
+    validates :grade, presence: true, on: :grade
+    validates :other_grade, presence: true, if: :grade_is_other?
+    validate :validate_grade_format, on: :grade, unless: :is_multiple_gcse? || :new_record?
+    validate :validate_grades, unless: :new_record?, on: :grades, if: :is_multiple_gcse?
```

Comment:
The validation here is not as clear as it could be. The next card will refactor this class so that English grades are saved in the `grade` column rather than using the `grades` column as a JSON blob.

That change will allow us to simplify this validation quite a lot.

Reply:
This will fix the bug I mentioned too 👍